PCRE2 support still requires PCRE1

libmodsecurity 2.9.7 compiles fine using --with-pcre2 only when both pcre1 and pcre2 dev packages are installed, which is wrong...
Moreover, the resulting binary links against both libraries... which is further wrong.

ldd mod_security2.so |grep pcre
ldd: warning: you do not have execution permission for `./mod_security2.so'
	libpcre.so.1 => not found
	libpcre2-8.so.0 => /usr/lib64/libpcre2-8.so.0 (0x0000ffffb9194000)
        ...
If only pcre2 dev package is installed, it fails at ./configure --with-pcre2 saying:

checking for libpcre config script... no
configure: *** pcre library not found.
configure: error: pcre library is required

In other words, --with-pcre2 still requires pcre1 dev.
This question is the same as the [previous one](https://github.com/SpiderLabs/ModSecurity/issues/2750),however, this issue is not resolved in v2.9.7.

--- a/build/find_pcre.m4
+++ b/build/find_pcre.m4
@@ -21,69 +21,73 @@ AC_ARG_WITH(
     [test_paths="${with_pcre}"],
     [test_paths="/usr/local/libpcre /usr/local/pcre /usr/local /opt/libpcre /opt/pcre /opt /usr"])
 
-AC_MSG_CHECKING([for libpcre config script])
-
-dnl # Determine pcre lib directory
-if test -z "${with_pcre}"; then
-    test_paths="/usr/local/pcre /usr/local /usr"
+if test "x${with_pcre2}" != "x" && test "x${with_pcre2}" != "xno"; then
+    AC_MSG_NOTICE([pcre2 specified; omitting check for pcre])
 else
-    test_paths="${with_pcre}"
-fi
+    AC_MSG_CHECKING([for libpcre config script])
 
-for x in ${test_paths}; do
-    dnl # Determine if the script was specified and use it directly
-    if test ! -d "$x" -a -e "$x"; then
-        PCRE_CONFIG=$x
-        pcre_path="no"
-        break
+    dnl # Determine pcre lib directory
+    if test -z "${with_pcre}"; then
+        test_paths="/usr/local/pcre /usr/local /usr"
+    else
+        test_paths="${with_pcre}"
     fi
 
-    dnl # Try known config script names/locations
-    for PCRE_CONFIG in pcre-config; do
-        if test -e "${x}/bin/${PCRE_CONFIG}"; then
-            pcre_path="${x}/bin"
+    for x in ${test_paths}; do
+        dnl # Determine if the script was specified and use it directly
+        if test ! -d "$x" -a -e "$x"; then
+            PCRE_CONFIG=$x
+            pcre_path="no"
             break
-        elif test -e "${x}/${PCRE_CONFIG}"; then
-            pcre_path="${x}"
+        fi
+
+        dnl # Try known config script names/locations
+        for PCRE_CONFIG in pcre-config; do
+            if test -e "${x}/bin/${PCRE_CONFIG}"; then
+                pcre_path="${x}/bin"
+                break
+            elif test -e "${x}/${PCRE_CONFIG}"; then
+                pcre_path="${x}"
+                break
+            else
+                pcre_path=""
+            fi
+        done
+        if test -n "$pcre_path"; then
             break
-        else
-            pcre_path=""
         fi
     done
-    if test -n "$pcre_path"; then
-        break
-    fi
-done
 
-if test -n "${pcre_path}"; then
-    if test "${pcre_path}" != "no"; then
-        PCRE_CONFIG="${pcre_path}/${PCRE_CONFIG}"
+    if test -n "${pcre_path}"; then
+        if test "${pcre_path}" != "no"; then
+            PCRE_CONFIG="${pcre_path}/${PCRE_CONFIG}"
+        fi
+        AC_MSG_RESULT([${PCRE_CONFIG}])
+        PCRE_VERSION="`${PCRE_CONFIG} --version`"
+        if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre VERSION: $PCRE_VERSION); fi
+        PCRE_CFLAGS="`${PCRE_CONFIG} --cflags`"
+        if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre CFLAGS: $PCRE_CFLAGS); fi
+        PCRE_LDADD="`${PCRE_CONFIG} --libs`"
+        if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre LDADD: $PCRE_LDADD); fi
+        PCRE_LD_PATH="/`${PCRE_CONFIG} --libs | cut -d'/' -f2,3,4,5,6 | cut -d ' ' -f1`"
+        if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre PCRE_LD_PATH: $PCRE_LD_PATH); fi
+    else
+        AC_MSG_RESULT([no])
     fi
-    AC_MSG_RESULT([${PCRE_CONFIG}])
-    PCRE_VERSION="`${PCRE_CONFIG} --version`"
-    if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre VERSION: $PCRE_VERSION); fi
-    PCRE_CFLAGS="`${PCRE_CONFIG} --cflags`"
-    if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre CFLAGS: $PCRE_CFLAGS); fi
-    PCRE_LDADD="`${PCRE_CONFIG} --libs`"
-    if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre LDADD: $PCRE_LDADD); fi
-    PCRE_LD_PATH="/`${PCRE_CONFIG} --libs | cut -d'/' -f2,3,4,5,6 | cut -d ' ' -f1`"
-    if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre PCRE_LD_PATH: $PCRE_LD_PATH); fi
-else
-    AC_MSG_RESULT([no])
+
+    AC_SUBST(PCRE_CONFIG)
+    AC_SUBST(PCRE_VERSION)
+    AC_SUBST(PCRE_CPPFLAGS)
+    AC_SUBST(PCRE_CFLAGS)
+    AC_SUBST(PCRE_LDFLAGS)
+    AC_SUBST(PCRE_LDADD)
+    AC_SUBST(PCRE_LD_PATH)
+
+    if test -z "${PCRE_VERSION}"; then
+        AC_MSG_NOTICE([*** pcre library not found.])
+    else
+        AC_MSG_NOTICE([using pcre v${PCRE_VERSION}])
+        ifelse([$1], , , $1) 
+    fi 
 fi
-
-AC_SUBST(PCRE_CONFIG)
-AC_SUBST(PCRE_VERSION)
-AC_SUBST(PCRE_CPPFLAGS)
-AC_SUBST(PCRE_CFLAGS)
-AC_SUBST(PCRE_LDFLAGS)
-AC_SUBST(PCRE_LDADD)
-AC_SUBST(PCRE_LD_PATH)
-
-if test -z "${PCRE_VERSION}"; then
-    AC_MSG_NOTICE([*** pcre library not found.])
-else
-    AC_MSG_NOTICE([using pcre v${PCRE_VERSION}])
-    ifelse([$1], , , $1) 
-fi 
 ])

